### ISSUE TITLE
the issue of visualizing results

### ISSUE THREAD

**Issue by XiaoW11** [RenYurui/Neural-Texture-Extraction-Distribution#30]
Hello
Your work is excellent, thank you for your open source, but when I started training code from scratch, I always reported errors during the visualization phase. After troubleshooting based on the error, it was found that it was the attent2image function

def attn2image(source_softmax, seg_query, input_image):
    target_list = []
    image_size = input_image.shape[2:]
    for softmax, query in zip(source_softmax, seg_query):
        b, num_label, h, w = query.shape
        input_resize = F.interpolate( input_image, (h,w) )
        input_resize = input_resize.view(b, -1, h*w)
        extracted = torch.einsum('bkm,bvm->bvk', softmax, input_resize)
        query = F.softmax(query.view(b, num_label, -1), 1)
        estimated_target = torch.einsum('bkm,bvk->bvm', query, extracted)
        estimated_target = estimated_target.view(b, -1, h, w)
        target_list.append(F.interpolate(estimated_target, image_size))
    target_gen = torch.cat(target_list, 3)
    return target_gen
    When doing torch. cat, the target_list, After calculation, the list is still an empty list, so I suspect that during visualization, the code
    
    attn_image=attn2image（info['book_softmax']，info['semantic_distribution']，input_image）
    nfo['book_softmax']，info['semantic_distribution']    is Still the empty value.
    
    But I have checked all the training processes and did not find any problems. May I ask if you encountered this problem during the experiment and how did you solve it
    Thank you！

**Comment by XiaoW11** [comment]
I didn't read the code carefully,just replace thr info['hook_softmax']  with ifo['extraction_softmax']
